### ISSUE TITLE
GEOS-Chem 14.3.0. Cannot get pointer to HEMCO field CH4_OIL.

### ISSUE THREAD

**Issue by zion0210** [geoschem/geos-chem#2202]
### Name and Institution (Required)

Name: zion0210
Institution: ASU

### Confirm you have reviewed the following documentation

- [ ] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/help-and-reference/SUPPORT.html)
- [ ] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [ ] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [ ] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

Hello, GEOS-Chem developers and users!

I ran different cases of the GEOS-Chem 14.3.0.
Also, I set the horizontal grid with 4.0x5.0 deg resolution and the Z grid with 47 dimensions.
CO2 - everything good,
CH4 - everything good,
Aerosols - everything good,
Carbon - the following error:
`GEOS-Chem ERROR: Cannot get pointer to HEMCO field CH4_OIL -> at Emiss_Carbon_Gases (in module GeosCore/carbon_gases_mod.F90)`

Previously, this bug has been noted in discussion [#2008](tg://search_hashtag?hashtag=2008) for v14.2.1.

Will I need to use an older version of the GEOS-Chem for carbon case?
Or can you suggest another solution to my problem?

Best regards,
zion0210

**Comment by msulprizio** [comment]
Hi @zion0210. We have been able to run the carbon simulation with GEOS-Chem 14.3.0 without issue. Can you confirm that you are using the correct versions of the GEOS-Chem (14.3.0) and HEMCO (3.8.0) submodules? Can you also confirm that you have created a new run directory after updating to 14.3.0 and are not using a run directory from a previous version? If that all looks fine, please upload your log file(s) here so we can have a closer look.

**Comment by zion0210** [comment]
Hello, @msulprizio!

I used the GEOS-Chem (14.3.0) and the HEMCO (3.8.0).

Today, I was clone code of the GEOS-Chem agane.

`git clone --recurse-submodules https://github.com/geoschem/GCClassic.git GCClassic_14.3.0`

After, I created run directory.

`./createRunDir.sh`

In this process I choose:
```
-----------------------------------------------------------
Choose simulation type:
-----------------------------------------------------------
   1. Full chemistry
   2. Aerosols only
   3. CH4
   4. CO2
   5. Hg
   6. POPs
   7. Tagged CH4
   8. Tagged CO
   9. Tagged O3
  10. TransportTracers
  11. Trace metals
  12. Carbon
>>> 12
-----------------------------------------------------------
Do you wish to use a single advected species?
-----------------------------------------------------------
  1. Use all species
  2. Use CH4 only
  3. Use CO2 only
  4. Use CO only
  5. Use OCS only
>>> 3
```

After, I made the code build.

```
cmake ../CodeDir -DRUNDIR=..
make -j 
make install
```
Then, I made dryrun.

```
./gcclassic  --dryrun > log.dryrun
./download_data.py log.dryrun  washu
```

However, I had problem this native Restarts/GEOSChem.Restart.20190101_0000z.nc4.

```
HEMCO: Opening ./Restarts/GEOSChem.Restart.20190101_0000z.nc4
 
HEMCO ERROR: Cannot find field SPC_CH2I2. Please check variable name in the config. file
 
HEMCO ERROR: Error encountered in routine HCOIO_Read!
 
HEMCO ERROR: Error in HCOIO_DATAREAD called from HEMCO ReadList_Fill: SPC_CH2I2
 --> LOCATION: ReadList_Fill (HCO_ReadList_Mod.F90)
 
HEMCO ERROR: Error in called ReadList_Fill (7) from HEMCO ReadList_Read
 --> LOCATION: ReadList_Read (HCO_ReadList_Mod.F90)
 Error in ReadList_Read called from hco_run
===============================================================================
GEOS-Chem ERROR: Error encountered in "HCO_Run"!
 -> at HCOI_GC_Run (in module GeosCore/hco_interface_gc_mod.F90)

THIS ERROR ORIGINATED IN HEMCO!  Please check the HEMCO log file for 
additional error messages!
===============================================================================
```

For this reason, I used full chemistry restart file.

```
********************************************
* B e g i n   T i m e   S t e p p i n g !! *
********************************************

---> DATE: 2019/01/01  UTC: 00:00
 HEMCO already called for this timestep. Returning.
===============================================================================
GEOS-Chem ERROR: Cannot get pointer to HEMCO field CH4_OIL
 -> at Emiss_Carbon_Gases (in module GeosCore/carbon_gases_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Emiss_Carbon_Gases"!
 -> at Emissions_Run (in module GeosCore/emissions_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Emissions_Run"! after drydep!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...
```

I attached config file and run log to this message.

[log+conf_gc_14.tar.gz](https://github.com/geoschem/geos-chem/files/14622659/log%2Bconf_gc_14.tar.gz)

Best regards,
zion0210

**Comment by yantosca** [comment]
Thanks for providing the log file @zion0210.  This error:
```console
HEMCO ERROR: Error in HCOIO_DATAREAD called from HEMCO ReadList_Fill: SPC_CH2I2
 --> LOCATION: ReadList_Fill (HCO_ReadList_Mod.F90)
```
indicates that the executable was not built with `cmake -DMECH=carbon`.   Try rebuilding with this option and then the run should proceed.

**Comment by zion0210** [comment]
Hello

I rebuilt the code:

```
make clean
cmake ../CodeDir -DMECH=carbon -DRUNDIR=..
make -j 
make install
```

I using native Restarts/GEOSChem.Restart.20190101_0000z.nc4 and GC reads all arrays which are necessary for start of the modeling.

However, I got the error again.

```
Successfully updated diagnostics: EmisCO2_BiomassBurn (counter:  1)
HEMCO: Leaving HCOX_GFED_Run (HCOX_GFED_MOD.F90) ( 2)
HEMCO: Leaving HCOX_RUN (HCOX_DRIVER_MOD.F90) ( 1)
===============================================================================
GEOS-Chem ERROR: Cannot get pointer to HEMCO field CH4_OIL
 -> at Emiss_Carbon_Gases (in module GeosCore/carbon_gases_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Emiss_Carbon_Gases"!
 -> at Emissions_Run (in module GeosCore/emissions_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Emissions_Run"! after drydep!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...

-------------------------------------------------------------------------------
HEMCO 3.8.0 FINISHED.
      Warnings: *
-------------------------------------------------------------------------------
```
[log.test2019.txt](https://github.com/geoschem/geos-chem/files/14663975/log.test2019.txt)

Best,
zion0210

**Comment by zion0210** [comment]
Hello!

I tried different configurations for GC 14.3.0.
And here's what I got. I set type of the simulation to "carbon".
After, I chose in the step "Do you wish to use a single advected species?" following options:
1. "Use all species" - test simulation was good,
2. "Use CH4 only" - test simulation was good,
3. "Use CO2 only" -  error: "Cannot get pointer to HEMCO field CH4_OIL"

I made the calculations for CO2 I needed only with the option 1.

Have you fixed the bug with the "CO2 only" option in the new version of GC (v14.3.1)?

Best,
zion0210

**Comment by msulprizio** [comment]
This is now resolved in the `dev/14.4.0` branch with the merge of https://github.com/geoschem/geos-chem/pull/1992.

There were a few missing logical brackets in HEMCO_Config.rc that caused issues when CH4 data was turned off.  You can either switch to this development branch or use the corrected [HEMCO_Config.rc.carbon](https://github.com/geoschem/geos-chem/blob/dev/14.4.0/run/GCClassic/HEMCO_Config.rc.templates/HEMCO_Config.rc.carbon) file.
